server/routes/visitJourney/dateAndTime.ts

```diff
@@ -102,10 +104,20 @@ export default class DateAndTime {
 
     visitSessionData.allowOverBooking = false // intentionally reset when returning to date and time page
 
+    const { validationError } = visitSessionData
+
+    let validationMessage = ''
+    if (validationError === 'APPLICATION_INVALID_NON_ASSOCIATION_VISITS') {
+      validationMessage = `${prisonerName} now has a non-association on ${format(visitSessionData.visitSlot.startTimestamp, 'd MMMM')}.`
+    } else if (validationError === 'APPLICATION_INVALID_VISIT_ALREADY_BOOKED') {
+      validationMessage = `${prisonerName} now has another visit at ${format(parseISO(visitSessionData.visitSlot.startTimestamp), 'h:mmaaa').replace(':00', '')} on ${format(visitSessionData.visitSlot.startTimestamp, 'd MMMM')}.`
+    }
+
```

Comment:
This kind of thing could potentially go in a utility function so determining message value for various conditions can be tested a lot more simply than at this level. Fine for now though as there are (currently!) only two options - and we'll be completely rewriting this page soon to do calendar view...

You might be able to simplify this code though with some of the functions in https://github.com/ministryofjustice/book-a-prison-visit-staff-ui/blob/3c829512d72d7b46152f52289b82aca754ce378a/server/utils/utils.ts. E.g. `prisonerTimePretty()` (badly named...)